python/ray/tune/tests/test_checkpoint_manager.py

```diff
@@ -104,6 +104,31 @@ def testBestCheckpoints(self):
         for i in range(len(best_checkpoints)):
             self.assertEqual(best_checkpoints[i].value, i + 12)
 
+    def testBestCheckpointsWithNan(self):
+        """
+        Tests that checkpoints with nan priority are handled correctly.
+        """
+        keep_checkpoints_num = 2
+        checkpoint_manager = self.checkpoint_manager(keep_checkpoints_num)
+        checkpoints = [
+            _TuneCheckpoint(
+                _TuneCheckpoint.PERSISTENT, None, self.mock_result(float("nan"), i)
+            )
+            for i in range(2)
+        ]
+        checkpoints += [
+            _TuneCheckpoint(_TuneCheckpoint.PERSISTENT, 3, self.mock_result(0, 3))
+        ]
+        random.shuffle(checkpoints)
+
+        for checkpoint in checkpoints:
+            checkpoint_manager.on_checkpoint(checkpoint)
+
+        best_checkpoints = checkpoint_manager.best_checkpoints()
+        self.assertEqual(len(best_checkpoints), keep_checkpoints_num)
+        self.assertEqual(best_checkpoints[0].value, None)
+        self.assertEqual(best_checkpoints[1].value, 3)
```

Comment:
Nit: Add a comment explaining that `checkpoint_manager.best_checkpoints()` has the worst checkpoints first.